test/test2.js

```diff
@@ -0,0 +1,231 @@
+            }
+          ]
+        }
+      },
+      afterInit: function(callback) {
+        assert(apos.modules['apostrophe-personas']);
+        // Should NOT have an alias!
+        assert(!apos.personas);
+        return callback(null);
+      },
+      afterListen: function(err) {
+        assert(!err);
+        done();
+      }
+    });
+  });
+
+  it('2. Test initial page load', function(done) {
+    const opts = {
+      url: basePath,
+      jar: j,
+      method: 'GET',
+      headers: {
+        Referrer: basePath
+      }
+    };
+
+    request(opts, (err, res) => {
+      assert(!err);
+      assert(res.statusCode === 200, 'req success');
+      assert(res.toJSON().body.indexOf('ello world') >= 0, 'Has default apostrophe homepage');
+      done();
+    });
+  });
+
+  it('3. Passing persona as query should set persona on session', function(done) {
+    const opts = {
+      url: basePath,
+      method: 'GET',
+      jar: j,
+      qs: {persona: '2r'},
+      headers: {
+        Referrer: basePath
+      }
+    };
+
+    request(opts, (err, res) => {
+      assert(!err);
+      assert(res.statusCode === 200, 'req success');
+      assert(res.toJSON().request.uri.pathname === '/2r/');
+      done();
+    });
+  });
+
+  it('4. Subsequent visit should load persona path', function(done) {
+    const opts = {
+      url: basePath,
+      method: 'GET',
+      jar: j,
+      headers: {
+        Referrer: basePath
+      }
+    };
+
+    request(opts, (err, res) => {
+      assert(!err);
+      assert(res.statusCode === 200, 'req success');
+      assert(res.req.path === '/2r/', 'Redirects to response specific path');
+      done();
+    });
+  });
+
+  it('5. New session loads generic home page', function(done) {
+    const opts = {
+      url: basePath,
+      method: 'GET',
+      jar: j1,
+
+      headers: {
+        Referrer: basePath
+      }
+    };
+
+    request(opts, (err, res) => {
+      assert(!err);
+      assert(res.statusCode === 200, 'req success');
+      assert(res.toJSON().body.indexOf('ello world') >= 0, 'Has default apostrophe homepage');
+      assert(res.req.path === '/', 'loads base path');
+      done();
+    });
+  });
+
+  it('6. Visit persona page sets persona session', function(done) {
+    const opts = {
+      url: basePath + 'tc-related',
+      method: 'GET',
+      jar: j1,
+      headers: {
+        Referrer: basePath
+      }
+    };
+
+    request(opts, (err, res) => {
+      assert(!err);
+      assert(res.statusCode === 200, 'req success');
+      done();
+    });
+  });
+
+  it('7. Subsequent request is forwarded to persona url', function(done) {
+    const opts = {
+      url: basePath,
+      method: 'GET',
+      jar: j1,
+      headers: {
+        Referrer: basePath
+      }
+    };
+
+    request(opts, (err, res) => {
+      assert(!err);
+      assert(res.statusCode === 200, 'req success');
+      assert(res.req.path === '/tc/', 'Redirects to response specific path');
```

Comment:
I think this is probably where you should verify that the body *doesn't* contain a widget for the *wrong* persona anymore, now that we're on a response specific path in a subsequent request. I think that is the only missing test of importance.